src/annotator/util/listener-collection.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @typedef Listener
+ * @prop {Window|Element|Document} eventTarget
+ * @prop {string} eventType
+ * @prop {(event: Event) => void} listener
+ */
+
+/**
+ * Utility that prevents forgetting to remove event listener.
+ */
+export function listenerCollection() {
```

Comment:
Was there a reason for making this a function that returns an object rather than a class? I know that in some parts of the sidebar (eg. services) this pattern is common, but that's mostly because it pre-dates ES classes. In modern code we tend to use actual classes partly because it makes it more obvious to the reader what they are looking at and partly because the tooling understands it better.

Reply:
It was a personal choice. I will transform the function into a class for consistency with the rest of the code.

Reply:
There is a however an advantage in the context of react functional components, it doesn't need to be wrap on a useEffect or useMemo;

/Users/esanzgar/Development/hypothesis/client/src/sidebar/components/ThreadList.js
  55:9  error  The 'listeners' object construction makes the dependencies of useEffect Hook (at line 171) change on every render. Move it inside the useEffect callback. Alternatively, wrap the initialization of 'listeners' in its own useMemo() Hook